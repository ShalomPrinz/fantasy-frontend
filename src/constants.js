export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 800;
