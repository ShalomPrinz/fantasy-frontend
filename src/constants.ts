export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 1000;
export const FULL_SCREEN_MIN_WIDTH = 1200;

// My Backend Endpoints
export const baseURL = "http://localhost:8080";
export const playersUrl = "/players/query";
export const usernamesUrl = "/users/query";
export const registerUrl = "/register";
export const userInfoUrl = "/user";
export const addPlayerUrl = "/user/addplayer";
export const removePlayerUrl = "/user/removeplayer";
export const acceptLeagueInviteUrl = "/accept/leagueinvite";
export const rejectLeagueInviteUrl = "/reject/leagueinvite";
