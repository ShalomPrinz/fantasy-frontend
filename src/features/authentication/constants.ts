/* API Endpoints */

export const registerUrl = "/register";
export const userInfoUrl = "/user";
