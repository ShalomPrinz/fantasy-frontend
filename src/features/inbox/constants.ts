/* API Endpoints */

export const acceptLeagueInviteUrl = "/accept/leagueinvite";
export const rejectLeagueInviteUrl = "/reject/leagueinvite";
