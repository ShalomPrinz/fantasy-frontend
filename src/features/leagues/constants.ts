export const LEAGUE_ID_LENGTH = 20;
export const LEAGUE_TEAM_LAYOUT_MAX_WIDTH = 680;

/* API Endpoints */

export const createLeagueUrl = "/newleague";
export const inviteLeagueMemberUrl = "/leagueinvite";
export const leagueInfoUrl = "/league";
