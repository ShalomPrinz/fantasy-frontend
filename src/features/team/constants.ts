export const FIELD_IMAGE_DEFAULT_WIDTH = 550;
export const FIELD_LAYOUT_MIN_WIDTH = 1000;

/* API Endpoints */

export const addPlayerUrl = "/user/addplayer";
export const removePlayerUrl = "/user/removeplayer";
