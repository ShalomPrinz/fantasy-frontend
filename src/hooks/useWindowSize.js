import { useEffect, useState } from "react";

function useWindowSize(callback) {
  const [size, setSize] = useState({ width: 0, height: 0 });
  useEffect(() => {
    function updateSize() {
      setSize({
        width: window.innerWidth,
        height: window.innerHeight
      });
      if (callback)
        callback();
    }
    window.addEventListener("resize", updateSize);
    updateSize();
    return () => window.removeEventListener("resize", updateSize);
  }, [callback]);
  return size;
}

export default useWindowSize;
