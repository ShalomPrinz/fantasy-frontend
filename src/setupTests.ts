// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import axios from "axios";
import { setupServer } from "msw/node";

import { handlers } from "../test";
import * as TeamContext from "./contexts/TeamContext";
import * as UserContext from "./features/authentication/contexts/UserContext";
import { LeagueInfo, Message, Player, Team, User } from "./types";

jest.setTimeout(10000);

axios.defaults.baseURL = "";

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

interface MockedUserContext {
  logout?: () => Promise<void>;
  state?: UserContext.UserState;
  user?: User;
}

interface MockedUser {
  inbox?: Message[];
  leagues?: LeagueInfo[];
  name?: string;
  players?: Player[];
}

export const getTestUser = (user?: MockedUser): User => ({
  id: "id",
  inbox: user?.inbox || [],
  leagues: user?.leagues || [],
  name: user?.name || "",
  team: new Team(user?.players || []),
});

export const mockUser = (context: MockedUserContext) => {
  const userContext = jest.spyOn(UserContext, "useUser");

  const { logout, state, user } = context;
  const mockedLogout = typeof logout !== "undefined" ? logout : async () => {};
  const mockedState =
    typeof state !== "undefined" ? state : UserContext.UserState.LOGGED_USER;
  const mockedUser = typeof user !== "undefined" ? user : getTestUser();

  userContext.mockImplementation(() => ({
    loading: false,
    login: async () => {},
    logout: mockedLogout,
    register: async () => {},
    state: mockedState,
    user: mockedUser,
    invalidate: async () => {},
  }));

  return userContext.mockRestore;
};

export const mockTeam = (team: Team) => {
  const teamContext = jest.spyOn(TeamContext, "useTeamState");
  teamContext.mockImplementation(() => team);
};

export * from "../test";
