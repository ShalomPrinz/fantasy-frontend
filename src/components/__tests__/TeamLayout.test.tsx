import React from 'react'

import { TeamLayout } from '../'
import { FIELD_LAYOUT_MIN_WIDTH } from '../../constants'
import useWindowSize from '../../hooks/useWindowSize'
import { getJSON, TestRendererJSON } from '../../setupTests'

let tree: TestRendererJSON = null;

beforeEach(() => {
    tree = null
})

jest.mock('hooks/useWindowSize')

const mockWindowSize = (width: number) =>
    // @ts-ignore: This is a mocked version of useWindowSize
    useWindowSize.mockReturnValue({ width: width })

describe('TeamLayout', () => {
    it('should render TeamList component', () => {
        const team = [{ id: 0, label: 'GK', players: [{id: 0, name: 'Ter Stegen', team: 'Barcelona'}] }]
    
        mockWindowSize(FIELD_LAYOUT_MIN_WIDTH)
        tree = getJSON(<TeamLayout team={team} />)
        
        expect(tree).toMatchSnapshot()
    })
    
    it('should render TeamLayout component', () => {
        const team = [{ id: 0, label: 'GK', players: [{id: 0, name: 'Ter Stegen', team: 'Barcelona'}] }]
        
        mockWindowSize(FIELD_LAYOUT_MIN_WIDTH + 1)
        tree = getJSON(<TeamLayout team={team} />)
        
        expect(tree).toMatchSnapshot()
    })
    
    it('should render bigger row margin if field width is higher than 600px', () => {
        const team = [{ id: 0, label: 'GK', players: [{id: 0, name: 'Ter Stegen', team: 'Barcelona'}] }]
        
        mockWindowSize(FIELD_LAYOUT_MIN_WIDTH + 1)
        React.useState = jest.fn(() => [601 , () => {}])
        tree = getJSON(<TeamLayout team={team} />)
        
        expect(tree).toMatchSnapshot()
    })
})
