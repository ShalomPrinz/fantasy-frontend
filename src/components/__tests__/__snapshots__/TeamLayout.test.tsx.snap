// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamLayout should render TeamLayout component 1`] = `
<div
  className="position-relative"
>
  <img
    alt="Team Field Background"
    className="w-100 rounded"
    src="field.png"
  />
  <div
    className="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
  >
    <div
      className="my-1 row"
    >
      <div
        className="mx-auto col"
      >
        <div
          className="text-center"
          style={
            Object {
              "width": "77px",
            }
          }
        >
          <img
            className=""
            src="Barcelona.png"
            style={
              Object {
                "width": "55px",
              }
            }
            title="Barcelona"
          />
          <div
            className="text-truncate bg-default rounded p-1 w-100"
          >
            Ter Stegen
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TeamLayout should render TeamList component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="w-50 col"
    >
      <h1
        className="centered-flex pt-4"
      >
         
        GK
         
      </h1>
      <div
        className="d-flex align-items-center ps-5"
      >
        <img
          className=""
          src="Barcelona.png"
          style={
            Object {
              "width": "60px",
            }
          }
          title="Barcelona"
        />
        <h4
          className="text-truncate ps-2 my-0"
        >
           
          Ter Stegen
           
        </h4>
      </div>
    </div>
  </div>
  <div
    className="row"
  />
</div>
`;

exports[`TeamLayout should render bigger row margin if field width is higher than 600px 1`] = `
<div
  className="position-relative"
>
  <img
    alt="Team Field Background"
    className="w-100 rounded"
    src="field.png"
  />
  <div
    className="position-absolute p-5 top-0 centered-flex flex-column h-100 overflow-hidden container"
  >
    <div
      className="my-3 row"
    >
      <div
        className="mx-auto col"
      >
        <div
          className="text-center"
          style={
            Object {
              "width": "84.14px",
            }
          }
        >
          <img
            className=""
            src="Barcelona.png"
            style={
              Object {
                "width": "60.1px",
              }
            }
            title="Barcelona"
          />
          <div
            className="text-truncate bg-default rounded p-1 w-100"
          >
            Ter Stegen
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
